### ISSUE TITLE
Choosing identifiers in the second metadata file

### ISSUE THREAD

**Issue by melpetera** [workflow4metabolomics/tool-table_merge#2]
The idea is to add an optional parameter to select a column in the other metadata file to replace the original identifiers with this. 

Example: a user wants to merge his sampleMetadata file with his dataMatrix, but wants ions to but named after the "namecustom" column in his variableMetadata file.

**Comment by lecorguille** [comment]
Crossref: https://github.com/workflow4metabolomics/xcms/issues/36 https://github.com/workflow4metabolomics/camera/issues/14

**Comment by melpetera** [comment]
After a good sleep, I was wondering whether it could be more relevent to make this functionality an independant module (a Data Handling one)? 

Since this concerns at least two modules, and since a user could want the same thing when going "out of Galaxy" without necessarily wanting to merge the data, wouldn't it be relevant to dedicate a module saying something like that: 
"Let me your dataMatrix and a metadata file containing an other column you want as an identifier (e.g. 'namecustom' in variableMetadata), and I will switch for your these two tables with your so wanted identifiers."

This could be used also if the user wants to replace an unsexy sample identifier like "ECxxx_TF43_RA2_23_MT" by "mice23" for example...

@yguitton @lecorguille @ethevenot @jfrancoismartin what do you think about that?

**Comment by lecorguille** [comment]
Hum ... so a tool to inverse two columns? ... Excel? 😉

**Comment by melpetera** [comment]
Not exactly... A tool to inverse two columns in metadata file PLUS replacing identifiers in the dataMatrix, and so without having to check by yourself that the two tables are sorted the same. 
I sometimes have to do that, on Excel, and I can assure you it is quite boring to do the check for the dataMatrix to be sure everything is allright (not long - if the order is the same- but boring). 
And if the order is not the same, then it becomes boring AND time consuming to be sure things are done well.

**Comment by melpetera** [comment]
During some Galaxy discussion time with analysts, it appears they do not want to have to do this themselves. 
Another thing is that actually it does not only concern 2 tools which can be modified to handle this question by adding advanced parameters, it concerns every future tools which would use identifiers in results display. (By the way, MCA would also need to be modified with advanced parameters.) 
Since it is something generic, and which can impact several tools, I am convinced it is better to simply have another little tool, that people can use -if they want- early in their workflows. This way, current and futur tools do not need to adapt and can just focus on what they are designed for. 
Any opinion others? @yguitton @ethevenot @jfrancoismartin

**Comment by melpetera** [comment]
Reviewing this old issue, the module mentionned has been developped and is available here: https://unh-pfem-gitlab.ara.inrae.fr/pfem-public/tool-idchoice

This tool is meant to be migrated to tools-metabolomics by the way.

Etienne J, can you confirm?

**Comment by etiennejls** [comment]
I do confirm

**Comment by melpetera** [comment]
Thanks @etiennejls then I will close this issue
